projects/wave-core-new/src/lib/layers/legend/legend-raster/raster-legend.component.ts

```diff
@@ -47,19 +47,27 @@ export class RasterLegendComponent implements OnInit, OnChanges {
     }
 
     ngOnChanges(changes: SimpleChanges): void {
-        if (changes.symbology) {
-            const symbology = changes.symbology.currentValue;
+        if (changes.layer) {
+            const symbology = changes.layer.currentValue.symbology;
             this.numberPipeParameters = RasterLegendComponent.calculateNumberPipeParameters(symbology.colorizer.breakpoints);
         }
     }
 
     private static calculateNumberPipeParameters(breakpoints: Array<ColorBreakpoint>): string {
```

Comment:
In my opinion, this kind of utility method is very good testable and we should do this.
You can look for *.spec* files for other utilities for getting examples.
You can just try out different scenarios and validate that the outcome matches your expectations.

Reply:
4cbe85d84d5c95a010fe888a072f8c78d00d6fc0